pkg/buildkite/jobs.go

```diff
@@ -15,87 +15,105 @@ import (
 	"github.com/buildkite/buildkite-mcp-server/pkg/trace"
 	"github.com/buildkite/go-buildkite/v4"
 	"github.com/mark3labs/mcp-go/mcp"
-	"github.com/mark3labs/mcp-go/server"
 	"github.com/rs/zerolog/log"
 	"go.opentelemetry.io/otel/attribute"
 )
 
-// withJobsPagination adds client-side pagination options to a tool with a max of 50 per page
-func withJobsPagination() mcp.ToolOption {
-	return func(tool *mcp.Tool) {
-		mcp.WithNumber("page",
-			mcp.Description("Page number for pagination (min 1)"),
-			mcp.Min(1),
-		)(tool)
-
-		mcp.WithNumber("perPage",
-			mcp.Description("Results per page for pagination (min 1, max 50)"),
-			mcp.Min(1),
-			mcp.Max(50),
-		)(tool)
-	}
+// GetJobsArgs struct for typed parameters
+type GetJobsArgs struct {
+	OrgSlug      string `json:"org"`
+	PipelineSlug string `json:"pipeline_slug"`
+	BuildNumber  string `json:"build_number"`
+	JobState     string `json:"job_state"`
+	IncludeAgent bool   `json:"include_agent"`
+	Page         int    `json:"page"`
+	PerPage      int    `json:"perPage"`
+}
+
+// GetJobLogsArgs struct for typed parameters
+type GetJobLogsArgs struct {
+	OrgSlug      string `json:"org"`
+	PipelineSlug string `json:"pipeline_slug"`
+	BuildNumber  string `json:"build_number"`
+	JobUUID      string `json:"job_uuid"`
 }
 
-func GetJobs(client BuildsClient) (tool mcp.Tool, handler server.ToolHandlerFunc) {
+func GetJobs(client BuildsClient) (tool mcp.Tool, handler mcp.TypedToolHandlerFunc[GetJobsArgs]) {
 	return mcp.NewTool("get_jobs",
 			mcp.WithDescription("Get all jobs for a specific build including their state, timing, commands, and execution details"),
-			mcp.WithString("org_slug",
+			mcp.WithString("org",
```

Comment:
#115 changed a bunch of these _to_ `org_slug` and removed the more obvious descriptions (including these one) - does this PR need to be rebased?